engine/src/cubos/engine/physics/plugin.cpp

```diff
@@ -0,0 +1,63 @@
+#include <cubos/core/ecs/system/query.hpp>
+
+#include <cubos/engine/physics/plugin.hpp>
+#include <cubos/engine/settings/plugin.hpp>
+#include <cubos/engine/transform/plugin.hpp>
+
+#include <glm/glm.hpp>
+
+using cubos::core::ecs::Query;
+using cubos::core::ecs::Read;
+using cubos::core::ecs::Write;
+
+using namespace cubos::engine;
+
+// Semi-implicit euler integration
+static void integrate(Position& position, PhysicsVelocity& physicsVelocity, const PhysicsMass& physicsMass, float duration)
+{
+    // We don’t integrate things with infinite mass.
+    if (physicsMass.inverseMass <= 0.0f) return;
+
+    // Work out the acceleration from the force.
+    glm::vec3 resultingAcceleration = physicsVelocity.acceleration;
+    resultingAcceleration += physicsVelocity.totalForce * physicsMass.inverseMass;
+
+    // Update linear velocity from the acceleration.
+    physicsVelocity.velocity += resultingAcceleration * duration;
+
+    // Impose drag.
```

Comment:
I'm not a fan of calling this drag, makes damping look optional.
Damping is a necessity since semi implicit euler will increase to total energy in the system overtime, violating conservation of energy. Normally it is a parameter of the simulation that is adjusted globally

Reply:
I agree, I'll make the damping global. We should keep drag for the component right? I think for most situations that covers what people will need for drag on an object.

Reply:
I don't think drag should be built into the core of our physics engine, users may set a global drag "feel" by adjusting the damping. If they then require different drags (e.g. water) a system to apply that force can be trivially written by them.